### DISCUSSION TITLE
Transposing via track names

### DISCUSSION THREAD

**Discussion by flop-sy** [akira0245/MidiBard#46] [Ideas]
I'm not sure if it's a feature already but is it possible to have transposing work via track names instead of file names? 

For example we could use the same format for file names being #<instrument><semitones>#.mid
track 1 being named #piano-12#
track 2 being named #lute+24#

Perhaps even simplifing it to +1/-1 for entire 12 semitones? i believe this is how bardmusicplayer handles transposing making it even more plug and play friendly

**Comment by akira0245** [comment]
New build should allow you use BMP style track name to switch instrument or transpose, this feature was added long time ago but I've always worried about it's stability under various situations so I did't put it on UI.
Track names like this will work.
![image](https://user-images.githubusercontent.com/33274390/151748669-c9ae8bc9-5cfe-4ce9-af0a-5dd6c2d2b9d2.png)
